### PR TITLE
Use dynamic ports and replace console logging with xunit logger in E2E tests

### PR DESCRIPTION
**Pull request by natemcmaster**
Addresses #209.

Use dynamic ports instead of fixed ports.
Switch from using console logging to using the xunit logger. Remove duplicated code.

### CODE REVIEW

File: test/FunctionalTests/E2ETestBase.cs [hunk 1]
```diff
@@ -15,83 +20,80 @@ public abstract class E2ETestBase
         private string _siteName;
         private ITestOutputHelper _output;
 
-        protected E2ETestBase(ITestOutputHelper output, string siteName, int basePort)
+        protected E2ETestBase(ITestOutputHelper output, string siteName)
         {
-            _output = output;
+            _output = output ?? throw new ArgumentNullException(nameof(output));
```
Comment (Eilon) [inline test/FunctionalTests/E2ETestBase.cs]: No throw expressions please.

File: test/FunctionalTests/E2ETestBase.cs [hunk 2]
```diff
@@ -141,5 +142,19 @@ protected virtual void AssertResponse(HttpResponseMessage response, string respo
         {
             throw new NotImplementedException("Must be overriden in derived types.");
         }
+
+        private static int GetNextPort()
```
Comment (Eilon) [inline test/FunctionalTests/E2ETestBase.cs]: I assume this code is copied from some other place? Is it worth putting it in one of our test helper classes?

File: test/FunctionalTests/XunitLogger.cs [hunk 3]
```diff
@@ -1,61 +0,0 @@
...
-// See License.txt in the project root for license information
-
-using System;
-using System.Linq;
-using Microsoft.Extensions.Logging;
-using Xunit.Abstractions;
-
-namespace EntropyTests
-{
-    public class XunitLogger : ILogger, IDisposable
```
Comment (Eilon) [inline test/FunctionalTests/XunitLogger.cs]: Nice.

File: test/FunctionalTests/E2ETestBase.cs (hunk 1 above)
Comment (natemcmaster) [inline test/FunctionalTests/E2ETestBase.cs]: Because of preference or technical reasons?

File: test/FunctionalTests/E2ETestBase.cs (hunk 2 above)
Comment (natemcmaster) [inline test/FunctionalTests/E2ETestBase.cs]: Yes copied. Probably not worth maintaining as an API. It's simple enough and it's just test code.

File: test/FunctionalTests/E2ETestBase.cs (hunk 1 above)
Comment (Eilon) [inline test/FunctionalTests/E2ETestBase.cs]: Engr team decided they are evil.

**Comment by natemcmaster** [comment]
:up: :date: no-mo-throw expressions

File: test/FunctionalTests/E2ETestBase.cs (hunk 2 above)
Comment (anurse) [inline test/FunctionalTests/E2ETestBase.cs]: Similar code actually already existed in the IntegrationTests package from Hosting. I've converted the Hosting version to be a copy of the Kestrel one and some of this code will be able to be removed once aspnet/Hosting#996 lands :).

**Pull request merged**